documentation/userguide/docs/shares.md

```diff
@@ -197,27 +200,93 @@ we receive an error. Once we have revoked access to all items we can delete the
 
 ![share](pictures/shares/shares_delete_unauth.png#zoom#shadow)
 
+## **AWS data sharing technical details**
+Here is a brief explanation of how each type of sharing mechanism is implemented in data.all. It is important to 
+understand what really happens in AWS when dealing with downstream integrations that will consume shared data.
+
+### S3 Bucket sharing
+In this type of share the permissions are granted to the IAM role specified in the request as principal. It 
+can be either a data.all team IAM role or an external role defined as consumption role.
+
+When a sharing task for an S3 Bucket is processed, data.all will:
+1. Update the S3 Bucket policy to add permissions to the principal IAM role
+2. Create/Update the IAM policy "Share policy" that grants IAM permissions to the requested S3 bucket and KMS key. Attach this policy to the principal IAM role.
+3. (If the Bucket is encrypted using a KMS key) Update the KMS Key policy to add permissions to the principal IAM role
+
+### Glue Table sharing
+In this type of share the permissions are granted to the IAM role specified in the request as principal. It 
+can be either a data.all team IAM role or an external role defined as consumption role.
+
+When a sharing task for a Glue Table is processed, data.all will:
+1. Create a Glue database in the target account with name of the original database plus the suffix `_shared`. This database will be re-used if other share requests for the same source databaser are processed for other principals in the same environment.
+2. (If the share is cross-account) Revoke IAMAllowedPrincipal permissions from the table to ensure Lake Formation is used in the management of the table access and update LakeFormation to use Version 3 if not already >=3
+3. Grant Lake Formation permissions the original database and table to the IAM principals in the target. If the share is cross account this step will create a RAM invitation that needs to be accepted.
+4. Create resource link table from the original table in the `_shared` database in the target account
+5. Grant Lake Formation permissions to the resource link table to the IAM principals.
```

Comment:
`Grant Lake Formation permissions to the resource link table for the IAM principal.`